tenseal/tensors/ckksvector.cpp

```diff
@@ -90,6 +90,11 @@ CKKSVector& CKKSVector::add_inplace(CKKSVector to_add) {
         throw invalid_argument("can't add vectors of different sizes");
     }
 
+    if (this->context->is_auto_mod_switch() &&
```

Comment:
Maybe the if condition can be separated in a dedicated function/method, something like `should_set_to_same_mod`.

Reply:
I think it's more explicit to leave it like this, isn't it? what's the advantage of separating it?

Reply:
Just because it's a similar condition in multiple places, that's all

Reply:
I added an inline function (thought it may be better inline), what do you think?

Reply:
Yes, I think it is easier to follow.

Maybe the check can be moved to `set_to_same_mod` completely.

Reply:
`set_to_same_mod` should do its job unconditionally of `auto_mod_switch` (as used in utils/matrix_ops.h) and therefore should be kept outside